static/js/shared/data_fetcher.ts

```diff
@@ -387,108 +146,112 @@ function computePerCapita(
+      places: places,
+      statVars: statVars,
+    })
+    .then((resp) => {
+      return resp.data;
     });
+  let denomDataPromise: Promise<StatApiResponse>;
+  if (denomStatVars.length > 0) {
+    denomDataPromise = axios
+      .post(`/api/stats`, {
+        places: places,
+        statVars: denomStatVars,
+      })
+      .then((resp) => {
+        return resp.data;
+      });
+  } else {
+    denomDataPromise = Promise.resolve({});
   }
 
-  const numStatsVars = statsVars.length;
-  let dcidParams = `?`;
+  let placeParams = `?`;
   for (const place of places) {
-    dcidParams += `&dcid=${place}`;
+    placeParams += `&dcid=${place}`;
   }
-  const apiDataPromises: Promise<AxiosResponse<StatApiResponse>>[] = [];
-  for (const statsVar of statsVars) {
-    apiDataPromises.push(axios.get(`/api/stats/${statsVar}${dcidParams}`));
-  }
-  if (perCapita) {
-    apiDataPromises.push(
-      axios.get(`/api/stats/${TOTAL_POPULATION_SV}${dcidParams}`)
-    );
-  } else {
-    for (const denom of denominators) {
-      apiDataPromises.push(axios.get(`/api/stats/${denom}${dcidParams}`));
-    }
-  }
-  const displayNamesPromise: Promise<AxiosResponse<
-    DisplayNameApiResponse
-  >> = axios.get(`/api/place/displayname${dcidParams}`);
+  const displayNamesPromise: Promise<DisplayNameApiResponse> = axios
+    .get(`/api/place/displayname${placeParams}`)
+    .then((resp) => {
+      return resp.data;
+    });
 
   // create list of promises containing apiDataPromises followed by displayNamesPromise
-  const apiPromises: Promise<AxiosResponse<any>>[] = [];
-  apiPromises.push(...apiDataPromises);
-  apiPromises.push(displayNamesPromise);
+  const apiPromises: Promise<StatApiResponse | DisplayNameApiResponse>[] = [
+    statDataPromise,
+    denomDataPromise,
+    displayNamesPromise,
+  ];
 
   return Promise.all(apiPromises).then((allResp) => {
-    const result = new StatsData(places, statsVars, [], {}, "");
+    const result = new StatData(places, statVars, [], {}, "");
     const numOccurencesPerDate: { [key: string]: number } = {};
     const numStatVarsPerPlace: { [key: string]: number } = {};
-    const displayNameMapping =
-      allResp.length > 0 && allResp[allResp.length - 1].data
-        ? allResp[allResp.length - 1].data
-        : {};
-    for (let i = 0; i < numStatsVars; i++) {
-      const sv = statsVars[i];
-      result.data[sv] = allResp[i].data;
-      if (perCapita) {
-        result.data[sv] = computePerCapita(
-          allResp[i].data,
-          allResp[numStatsVars].data,
-          scaling
-        );
-      } else if (denominators.length) {
-        result.data[sv] = computePerCapita(
-          allResp[i].data,
-          allResp[i + numStatsVars].data,
-          scaling
-        );
+    const statResp = allResp[0] as StatApiResponse;
+    const denomResp = allResp[1] as StatApiResponse;
+    const displayNameMapping = (allResp[2]
+      ? allResp[2]
+      : {}) as DisplayNameApiResponse;
```

Comment:
why is this special handling required only for display names? could there be empty response for the stat api calls too?

Reply:
Double checked this could not be empty, so removing the check. Stat resp will always return non-null value too.